src/main/java/seedu/address/logic/commands/StackUndoRedo.java

```diff
@@ -0,0 +1,95 @@
+package seedu.address.logic.commands;
+
+import java.util.LinkedList;
+
+public class StackUndoRedo {
+
+    private LinkedList<UndoableCommand> undoStack;
+    private LinkedList<UndoableCommand> redoStack;
+
+    /**
+     * Pushes {@code command} onto the undo-stack if it is of type {@code UndoableCommand}. Clears the redo-stack
+     * if {@code command} is not of type {@code UndoCommand} or {@code RedoCommand}.
+     */
+    public StackUndoRedo() {
+        undoStack = new LinkedList<>();
+        redoStack = new LinkedList<>();
+    }
+
+    /**
+     * Pushes {@code command} onto the undo-stack if it is of type {@code UndoableCommand}. Clears the redo-stack
+     * if {@code command} is not of type {@code UndoCommand} or {@code RedoCommand}.
+     */
+    public StackUndoRedo(StackUndoRedo otherUndoRedoStack) {
+        undoStack = otherUndoRedoStack.undoStack;
+        redoStack = otherUndoRedoStack.redoStack;
+    }
+
+    /**
+     * Pushes {@code command} onto the undo-stack if it is of type {@code UndoableCommand}. Clears the redo-stack
+     * if {@code command} is not of type {@code UndoCommand} or {@code RedoCommand}.
+     */
+    public void push(Command command) {
+        if (!(command instanceof UndoableCommand)) {
+            return;
+        }
+
+        if (!(command instanceof UndoCommand) && !(command instanceof RedoCommand)) {
+            redoStack.clear();
```

Comment:
Not sure if you really need to clear it or not. Will confirm when the other review stuff has been fixed.

Reply:
Don't think this is necessary. Doesn't seem to be any case where this will ever get executed.